atspi-common/src/object_ref.rs

```diff
@@ -98,3 +137,134 @@ impl From<ObjectRef> for zvariant::Structure<'_> {
 		(obj.name, obj.path).into()
 	}
 }
+
+#[cfg(feature = "zbus")]
+impl TryFrom<&zbus::message::Header<'_>> for ObjectRef {
+	type Error = crate::AtspiError;
+	fn try_from(header: &zbus::message::Header) -> Result<Self, Self::Error> {
+		let path = header.path().ok_or(crate::AtspiError::MissingPath)?;
+		let owned_path: OwnedObjectPath = path.clone().into();
+
+		let sender: UniqueName<'_> = header.sender().ok_or(crate::AtspiError::MissingName)?.into();
+		let name: OwnedUniqueName = sender.to_owned().into();
+
+		Ok(ObjectRef { name, path: owned_path })
+	}
+}
+
+#[cfg(test)]
+mod test {
+	use zvariant::Value;
+
+	use crate::{object_ref::ObjectRefBorrowed, ObjectRef};
+
+	#[test]
+	fn test_accessible_from_dbus_ctxt_to_object_ref() {
+		use zvariant::serialized::Context;
+		use zvariant::{to_bytes, Value, LE};
+
+		let acc = ObjectRef::default();
+		let ctxt = Context::new_dbus(LE, 0);
+		let acc_value: Value<'_> = acc.into();
+		let data = to_bytes(ctxt, &acc_value).unwrap();
+		let (value, _) = data.deserialize::<Value>().unwrap();
+		let accessible: ObjectRef = value.try_into().unwrap();
+
+		assert_eq!(accessible.name.as_str(), ":0.0");
```

Comment:
If you are going to test this, make sure the strings are `const` and used both here and in the `Default` impl. Surprise tests failing when changing a default would be irritating.